cunumeric/random/random.py

```diff
@@ -924,6 +1015,40 @@ def pareto(
     return generator.get_static_generator().pareto(a, size, dtype)
 
 
+def permutation(x: Union[int, ndarray]) -> ndarray:
+    """
+    Returns a permuted range or a permutation of an array.
+
+    Parameters
+    ----------
+    a : (int or ndarray): The range or the array to be shuffled.
+
+    Returns
+    -------
+    out: If `a` is an integer, it is permutation range between 0
+        and `a` - 1.
+        Otherwise, it is a permutation of `a`.
+
+    See Also
+    --------
+    numpy.random.permutation
+
+    Availability
+    --------
+    Multiple GPUs, Multiple CPUs
+    """
+    if isinstance(x, int):
+        count = x
+    else:
+        count = len(x)
+    key = uniform(0.0, 1.0, count)
+    indices = cunumeric.argsort(key)
+    if isinstance(x, int):
+        return indices
+    else:
+        return x[indices]
```

Comment:
nit suggestion:

```suggestion
    count = x if isinstance(x, int) else len(x)
    
    key = uniform(0.0, 1.0, count)
    indices = cunumeric.argsort(key)
    
    return indices if isinstance(x, int) else x[indices]
```

I advocate for ternaries when they are short because I do think it makes the overall structure easier to take in at a glance (here: compute a count, compute indices from count, return appropriate result)